scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -445,9 +450,10 @@ class FormatWriter(formatOps: FormatOps) {
       //
       // Remove the comma after b (before comment)
       case TrailingCommas.never
-          if left.is[Comment] && prev(formatToken).left.is[Comma] &&
+          if left.is[Comment] && prevFormatToken.left.is[Comma] &&
             right.is[CloseDelim] && isNewline =>
-        sb.deleteCharAt(sb.length - left.syntax.length - 2)
+        sb.deleteCharAt(
```

Comment:
I am wondering if it's possible to prevent printing the trailling comma in the first place instead of deleting the characters afterwards 🤔

Reply:
I've tried a bit when sketching the first implementation, but then I got too confused 😓. I agree it's not airtight, but at least now we're computing the index dynamically instead of hardcoding numbers...